### PR TITLE
F/amqp rework

### PR DESCRIPTION
**Pull request by Mararok**

### CODE REVIEW

File: src/SAREhub/Client/Amqp/BasicAmqpProcessConfirmStrategy.php [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+     * @var bool
+     */
+    private $requeueOnReject = true;
+
+    public function __construct()
+    {
+        $this->logger = new NullLogger();
+    }
+
+    public function confirm(AmqpChannelWrapper $channel, Message $orginalIn, Exchange $exchange)
```
Comment (kuba9392) [inline src/SAREhub/Client/Amqp/BasicAmqpProcessConfirmStrategy.php]: little typo in $orginalIn variable, should be $originalIn.

File: tests/unit/SAREhub/Client/Amqp/AmqpChannelWrapperTest.php [hunk 2]
```diff
@@ -2,121 +2,201 @@
...
+
+        $this->processConfirmStrategy = Mockery::mock(AmqpProcessConfirmStrategy::class);
+        $this->wrapper->setProcessConfirmStrategy($this->processConfirmStrategy);
+
+        $this->consumer = $this->createConsumer(AmqpConsumerOptions::newInstance()
+            ->setQueueName("test_queue")
+            ->setTag("test_consumer_tag")
+            ->setExclusive(true));
+
+        //$this->channel->allows(["basic_consume" => $this->consumer->getOptions()->getTag()]);
```
Comment (kuba9392) [inline tests/unit/SAREhub/Client/Amqp/AmqpChannelWrapperTest.php]: if this is unused then comment should be deleted

File: tests/unit/SAREhub/Client/Processor/PipelineTest.php [hunk 3]
```diff
@@ -90,48 +86,72 @@ public function testProcessWhenProcessorSetsOutThenNextProcessorGetIt()
...
+        $p2 = $this->createProcessor();
+        $p2->shouldNotReceive("process");
+        $this->pipeline->add($p2);
+
+        $this->pipeline->process($exchange);
+    }
+
     public function testToString()
     {
         $this->pipeline
```
Comment (kuba9392) [inline tests/unit/SAREhub/Client/Processor/PipelineTest.php:157]: are there any assertions?

Comment (Mararok) [reply tests/unit/SAREhub/Client/Processor/PipelineTest.php:157]: yes look at all file

**Review by kuba9392 — CHANGES_REQUESTED** [review]
waiting for changes described in comments

**Review by kuba9392 — APPROVED** [review]

**Pull request merged**